components/user_profile/user_profile.tsx

```diff
@@ -1,52 +1,60 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React, {PureComponent} from 'react';
 
+import {UserProfile as UserProfileType} from 'mattermost-redux/types/users';
+
 import {imageURLForUser, isMobile, isGuest} from 'utils/utils.jsx';
 
-import OverlayTrigger from 'components/overlay_trigger';
+import OverlayTrigger, {HideableOverlayTrigger} from 'components/overlay_trigger';
 import ProfilePopover from 'components/profile_popover';
 import BotBadge from 'components/widgets/badges/bot_badge';
 import GuestBadge from 'components/widgets/badges/guest_badge';
 
-export default class UserProfile extends PureComponent {
-    static propTypes = {
-        disablePopover: PropTypes.bool,
-        displayName: PropTypes.string,
-        displayUsername: PropTypes.bool,
-        hasMention: PropTypes.bool,
-        hideStatus: PropTypes.bool,
-        isBusy: PropTypes.bool,
-        isRHS: PropTypes.bool,
-        overwriteName: PropTypes.node,
-        overwriteIcon: PropTypes.node,
-        user: PropTypes.object,
-        userId: PropTypes.string,
-    };
-
-    static defaultProps = {
+export type UserProfileProps = {
+    displayName?: string,
+    isBusy?: boolean,
+    overwriteName?: React.ReactNode,
+    overwriteIcon?: React.ReactNode,
+    user?:UserProfileType,
+    userId: string,
+} & Partial<UserProfileDefaultProps>
```

Comment:
Why extend a partial type here? I think we generally do this elsewhere by including all the fields in the `Props` type and then defining `defaultProps` as a `Partial<Props>` instead of giving it its own type.

Reply:
Yeah, I wasn't 100% sure on how to do this. There seem to be many different practices around default props with TypeScript. I'll try to compare with other components in the same code base and make the necessary adjustments to follow the same approach. Thanks for catching this :)

Reply:
Applied the approach you suggested and it actually makes a lot of sense. Much simpler and readable. Thanks :)